src/components/metadataFormControls/DataElementsTransfer/useOptionsQuery.ts

```diff
@@ -0,0 +1,77 @@
+import { useDataQuery } from '@dhis2/app-runtime'
+import { useEffect, useMemo, useState } from 'react'
+import { SelectOption } from '../../../types'
+import { DataElement, Pager } from '../../../types/generated'
+
+type DataElementsQueryResult = {
+    dataElements: {
+        pager: Pager
+        dataElements: DataElement[]
+    }
+}
+
+const CATEGORY_COMBOS_QUERY = {
+    dataElements: {
+        resource: 'dataElements',
+        params: (variables: Record<string, string>) => {
+            const params = {
+                page: variables.page,
+                pageSize: 10,
+                fields: ['id', 'displayName'],
+                order: ['displayName'],
+            }
+
+            if (variables.filter) {
+                return {
+                    ...params,
+                    filter: `name:ilike:${variables.filter}`,
+                }
+            }
+
+            return params
+        },
+    },
+}
+
+export function useOptionsQuery() {
```

Comment:
I feel like this hook is duplicated a lot, and I think an abstraction like `useInfiniteDataQuery` would make this a bit cleaner. I attempted something similar here: https://github.com/dhis2/maintenance-app-beta/pull/367/files#diff-09b250214c7f359aeb6f4e48983d69b1d4cc09b437d6ba6cbfba0d86933acddf . 

However we don't have much time, and we could take a look at this after release. But I'm a bit worried about the amount of duplication that piles up.